blog/2018/12/22/use-carton-for-your-mojolicious-app-deployment/index.markdown

```diff
@@ -0,0 +1,155 @@
+
+But we can also list optional modules: you can support different databases, but users shouldn’t have to install MySQL-related modules if they want to use PostgreSQL.
+
+Here’s an example of `cpanfile`:
+
+    # Do not ask for a specific version
+    requires 'DateTime';
+    # Ask a specific version
+    requires 'Plack', '== 1.0';
+    # Ask a minimal version
+    requires 'Net::DNS', '>= 1.12';
+    # Or
+    requires 'Net::DNS', '1.12';
+    # Ask a maximal version
+    requires 'Locale::Maketext', '< 1.28';
+    # Give a range
+    requires 'Mojolicious', '>= 7.0, < 8.0';
+
+    # Optional modules
+    feature 'postgresql', 'PostgreSQL support' => sub {
+        requires 'Mojo::Pg';
+    };
+    feature 'mysql', 'MySQL support' => sub {
+        requires 'Mojo::mysql';
+    };
+    feature 'ldap', 'LDAP authentication support' => sub {
+        requires 'Net::LDAP';
+    };
+
+Cpanfile format can do more (recommended modules, requirements for a specific phase (`configure`, `test`…), using modules not published on CPAN…), but this is a post about Carton: I let you read cpanfile documentation 🙂
+
+Cpanfile can be used by [cpanminus](https://metacpan.org/pod/cpanm) or [Carton](https://metacpan.org/pod/Carton).
+
+Go to the directory containing your `cpanfile` and do:
+
+    cpanm --installdeps .
+
+*Et voilà !*
+
+Note that the modules in `features` has not been installed. You can install them with:
+
+    cpanm --installdeps . --with-feature postgresql
+
+Or, to install all `features` modules, but not the `mysql` one:
+
+    cpanm --installdeps . --with-all-features --without-feature mysql
+
+So, now, we can be sure that we have the good version of our application’s dependencies on the system.
+But what if we host other applications on that system, that have conflicting requirements?
+
+Cpanm is able to install modules in a specific folder (thank you, [local::lib](https://metacpan.org/pod/local::lib)), but wouldn’t it be convenient to install our dependencies in the directory of our application?
+We would always know where our dependencies are.
+
+## Here comes Carton
+
+[Carton](https://metacpan.org/pod/Carton) is Perl module dependency manager. Think `bundler` in Ruby. Think `npm` in Node.js.
+
+Like `npm` does, Carton install the dependencies in the directory of the application.
+
+### Deployment
+
+First, install Carton:
+
+    cpanm Carton
+
+Then, we can install our dependencies with:
+
+    carton install
+
+Our dependencies will be installed in a directory named `local`.
+But there is more: Carton will generate a `cpanfile.snapshot` file, containing the exact version of our dependencies, allowing us to enforce those exact version (ship it with your application).
+
+In our `cpanfile` example, we asked for a Mojolicious version superior or equal to 7.0 and inferior to 8.0.
```

Comment:
I'd use s/superior/greater than/ and s/inferior/less than/. Superior and inferior carry connotation of goodness.

Reply:
Bad translation from my french mind (`greater than` = `supérieur à` in french).

:+1: